pool/pool.go

```diff
@@ -56,6 +56,7 @@ type preExecutionResponse struct {
 	usedZkCounters       state.ZKCounters
 	isExecutorLevelError bool
 	isOOC                bool
+	OOCError             error
```

Comment:
Maybe it is cleaner if we change both isOOC and isOOG for errors en remove both booleans.

Reply:
Yes, I think this is a good idea and I'll try to do this refactoring